spv/envelope.go

```diff
@@ -42,39 +41,22 @@ func (e *Envelope) ParentTx(txID string) (*bt.Tx, error) {
 // Bytes takes an spvEnvelope struct and returns the serialised binary format.
 func (e *Envelope) Bytes() ([]byte, error) {
 	flake := make([]byte, 0)
-
-	// Binary format version 1
-	flake = append(flake, 1)
-
-	initialTx := map[string]*Envelope{
-		e.TxID: {
-			TxID:          e.TxID,
-			RawTx:         e.RawTx,
-			Proof:         e.Proof,
-			MapiResponses: e.MapiResponses,
-			Parents:       e.Parents,
-		},
-	}
-
-	err := serializeParents(initialTx, &flake, true)
+	flake = append(flake, 1) // Binary format version 1
+	err := serialiseInputs(e.Parents, &flake)
 	if err != nil {
-		fmt.Println(err)
+		return nil, err
 	}
 	return flake, nil
 }
 
-func serializeParents(parents map[string]*Envelope, flake *[]byte, root bool) error {
+func serialiseInputs(parents map[string]*Envelope, flake *[]byte) error {
```

Comment:
any reason why it's a pointer to and array of bytes instead of just the array of bytes there?

Reply:
I don't recall, I'll take a close look.